libs/novui/src/components/table/Table.tsx

```diff
@@ -0,0 +1,107 @@
+import { Table as ExternalTable } from '@mantine/core';
+import {
+  CellContext,
+  flexRender,
+  getCoreRowModel,
+  Row,
+  useReactTable,
+  createColumnHelper as _createColumnHelper,
+  ColumnDefTemplate,
+} from '@tanstack/react-table';
+import React, { useMemo } from 'react';
+import { CoreProps } from '../../types';
+
+import classes from './Table.styles';
+
+export type IRow<TRow extends object = {}> = Row<TRow>;
+
+/** Component used to render the contents of a cell */
+export type CellRendererComponent<TRow, TCellValue> = ColumnDefTemplate<CellContext<TRow, TCellValue>>;
+
+/** Helper for type-safe column definitions */
+export const createColumnHelper = _createColumnHelper;
+
+export interface ITableProps<TRow extends object> extends CoreProps {
+  columns?: any[];
+  data?: TRow[];
+  isLoading?: boolean;
+  pagination?: any;
+  noDataPlaceholder?: React.ReactNode;
+  loadingItems?: number;
+  hasMore?: boolean;
+  onRowClick?: (row: Row<TRow>) => void;
+  onRowSelect?: (row: Row<TRow>) => void;
+}
+
+/**
+ * Table component
+ *
+ */
+export function Table<TRow extends object>({
+  columns: userColumns,
+  data: userData,
+  isLoading = false,
+  noDataPlaceholder,
+  loadingItems = 10,
+  onRowClick,
+  onRowSelect,
+  ...props
+}: ITableProps<TRow>) {
+  const columns = useMemo(() => userColumns?.map((col) => ({ ...col })), [userColumns]);
+  const data = useMemo(() => (userData || [])?.map((row) => ({ ...row })), [userData]);
+  const fakeData = useMemo(() => Array.from({ length: loadingItems }).map((_, index) => ({ index })), [loadingItems]);
+
+  const table = useReactTable<TRow>({
+    columns,
+    data: isLoading ? (fakeData as TRow[]) : data,
+    getCoreRowModel: getCoreRowModel(),
+    // FIXME: remove this
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Was the intention to remove this one?‏